plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1022,8 +988,8 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
             // Remove the transaction log entry if the table creation fails
             try {
                 Path transactionLogLocation = getTransactionLogDir(new Path(handle.getLocation()));
-                FileSystem fs = hdfsEnvironment.getFileSystem(new HdfsContext(session), transactionLogLocation);
-                fs.delete(transactionLogLocation, true);
+                TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+                fileSystem.deleteFile(transactionLogLocation.toString());
```

Comment:
This call may likely fail because  `_delta_log` is a directory.
Note that previously `fs.delete(transactionLogLocation, true);` call was made.
Please consider using `deleteDirectory` method instead here.

BTW we could have also a test to simulate `CREATE TABLE` failure and make sure that the cleaning up is effective - it is quite annoying to get the message that the table already exists (due to the fact that previously the failed creation table left behind unwanted files).

Reply:
Thanks for your suggestion, I'm going to send another PR for `CREATE TABLE` failure.